.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,29 @@
+name: CodeQL
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '0 13 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: ${{ matrix.language }}
+      - name: Execute gradle build
+        run: ./gradlew --parallel --max-workers=4 clean assemble
```

Comment:
Any way to make the number of workers automatic?

Reply:
IIRC this was primarily for our older CI systems that were relatively under powered. This config is on all our builds currently so I'll leave it in this PR for consistency but we could experiment by removing this param as a followup.

```bash
--parallel                         Build projects in parallel. Gradle will attempt to determine the optimal number of executor threads to use.
```